test/fail_compilation/fail4082posix.d

```diff
@@ -0,0 +1,26 @@
+// DISABLED: win
+
+// POSIX only
+version (Posix) {}
+else static assert(0);
+
+/*
+TEST_OUTPUT:
+---
+fail_compilation/fail4082posix.d(9): Error: destructor `fail4082posix.Bar.~this` is not `nothrow`
+fail_compilation/fail4082posix.d(9): Error: `nothrow` function `fail4082posix.test2` may throw
+---
+*/
+
+#line 1
+struct Bar
+{
+    ~this() { throw new Exception(""); }
+}
+nothrow void test()
+{
+    test2(Bar.init);
+}
+nothrow void test2(Bar t)
+{
+}
```

Comment:
This seems to be the same as the windows one, with the exception of having `test`. Why the double tests ?

Reply:
I don't know. Nobody mentioned it in reviewing https://github.com/dlang/dmd/pull/10593 I presume there was a reason.

Reply:
I think that was the result of iterations (so that Windows could be selectively disabled). Can you merge them ?

Reply:
The lack of review doesn't mean it would have passed review, though

Reply:
> I don't know. Nobody mentioned it in reviewing #10593 I presume there was a reason.

Actually @ibuclaw did: https://github.com/dlang/dmd/pull/10593/commits/6cc40cffb03deb0fa44de3f49348af182f3c30f8#r352061726

But you fixed the issue, so this bit can be reverted, as there's no difference anymore.